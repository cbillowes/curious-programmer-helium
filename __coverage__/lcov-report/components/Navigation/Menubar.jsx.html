<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for components/Navigation/Menubar.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> / <a href="index.html">components/Navigation</a> Menubar.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>0/17</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/12</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/5</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>0/17</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53</td><td class="line-coverage quiet"><span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js"><span class="cstat-no" title="statement not covered" >import React, { Component } from "react"</span>
<span class="cstat-no" title="statement not covered" >import Link from "gatsby-link"</span>
<span class="cstat-no" title="statement not covered" >import navigation from "../../../data/Navigation"</span>
<span class="cstat-no" title="statement not covered" >import "./Menubar.scss"</span>
&nbsp;
class Menubar extends Component {
<span class="fstat-no" title="function not covered" >  constructor</span>(props) <span class="cstat-no" title="statement not covered" >{</span>
<span class="cstat-no" title="statement not covered" >    super(props)</span>
  }
&nbsp;
  getRootUrl = <span class="fstat-no" title="function not covered" >() =&gt; </span>{
<span class="cstat-no" title="statement not covered" >    if (typeof window !== "undefined" &amp;&amp; window.location) {</span>
      let url = <span class="cstat-no" title="statement not covered" >window.location.pathname</span>
<span class="cstat-no" title="statement not covered" >      if (url !== "/") {</span>
        let paths = <span class="cstat-no" title="statement not covered" >url.split("/")</span>
<span class="cstat-no" title="statement not covered" >        if (paths[1] === "blog") {</span>
<span class="cstat-no" title="statement not covered" >          return "/"</span>
        } else {
<span class="cstat-no" title="statement not covered" >          return `/${paths[1]}`</span>
        }
      }
<span class="cstat-no" title="statement not covered" >      return url</span>
    }
  }
&nbsp;
  isSelected = <span class="fstat-no" title="function not covered" >url</span> =&gt; {
<span class="cstat-no" title="statement not covered" >    return this.getRootUrl() === url ? "selected" : ""</span>
  }
&nbsp;
<span class="fstat-no" title="function not covered" >  render</span>() {
<span class="cstat-no" title="statement not covered" >    return (</span>
      &lt;nav
        className={`menubar ${this.props.active ? "active" : ""}`}
        onClick={this.props.toggle.bind(this)}
      &gt;
        &lt;ul&gt;
          {navigation.map(<span class="fstat-no" title="function not covered" >item</span> =&gt; {
<span class="cstat-no" title="statement not covered" >            return (</span>
              &lt;li
                key={`menubar-${item.url}`}
                className={`menu-item ${this.isSelected(item.url)}`}
              &gt;
                &lt;Link to={item.url}&gt;{item.name}&lt;/Link&gt;
              &lt;/li&gt;
            )
          })}
        &lt;/ul&gt;
      &lt;/nav&gt;
    )
  }
}
&nbsp;
export default Menubar</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sun Jan 06 2019 09:15:22 GMT+0200 (South Africa Standard Time)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
