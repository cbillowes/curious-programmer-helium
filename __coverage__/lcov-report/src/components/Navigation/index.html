<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/Navigation</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> src/components/Navigation
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">36.73% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>18/49</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">34.38% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>11/32</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">55.56% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>10/18</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">36.73% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>18/49</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file medium" data-value="Menubar.jsx"><a href="Menubar.jsx.html">Menubar.jsx</a></td>
	<td data-value="66.67" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 66%;"></div><div class="cover-empty" style="width:34%;"></div></div></td>
	<td data-value="66.67" class="pct medium">66.67%</td>
	<td data-value="12" class="abs medium">8/12</td>
	<td data-value="58.33" class="pct medium">58.33%</td>
	<td data-value="12" class="abs medium">7/12</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="5" class="abs high">5/5</td>
	<td data-value="66.67" class="pct medium">66.67%</td>
	<td data-value="12" class="abs medium">8/12</td>
	</tr>

<tr>
	<td class="file medium" data-value="Navigation.jsx"><a href="Navigation.jsx.html">Navigation.jsx</a></td>
	<td data-value="50" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 50%;"></div><div class="cover-empty" style="width:50%;"></div></div></td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="6" class="abs medium">3/6</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="40" class="pct low">40%</td>
	<td data-value="5" class="abs low">2/5</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="6" class="abs medium">3/6</td>
	</tr>

<tr>
	<td class="file low" data-value="Searchbar.jsx"><a href="Searchbar.jsx.html">Searchbar.jsx</a></td>
	<td data-value="22.58" class="pic low"><div class="chart"><div class="cover-fill" style="width: 22%;"></div><div class="cover-empty" style="width:78%;"></div></div></td>
	<td data-value="22.58" class="pct low">22.58%</td>
	<td data-value="31" class="abs low">7/31</td>
	<td data-value="20" class="pct low">20%</td>
	<td data-value="20" class="abs low">4/20</td>
	<td data-value="37.5" class="pct low">37.5%</td>
	<td data-value="8" class="abs low">3/8</td>
	<td data-value="22.58" class="pct low">22.58%</td>
	<td data-value="31" class="abs low">7/31</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sun Jan 06 2019 10:12:09 GMT+0200 (South Africa Standard Time)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
