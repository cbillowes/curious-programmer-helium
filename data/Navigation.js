module.exports = [
  {
    url: "/",
    name: "Blog"
  },
  {
    url: "/about",
    name: "About"
  },
  {
    url: "/archives",
    name: "Archives"
  },
  {
    url: "/tags",
    name: "Tags"
  },
  {
    url: "/credits",
    name: "Credits"
  },
  {
    url: "/privacy-policy",
    name: "Privacy Policy"
  },
  {
    url: "/license",
    name: "License"
  }
]
