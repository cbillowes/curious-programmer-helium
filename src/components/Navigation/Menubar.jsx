import React, { Component } from "react"
import Link from "gatsby-link"
import navigation from "../../../data/Navigation"
import "./Menubar.scss"

class Menubar extends Component {
  constructor(props) {
    super(props)
  }

  getRootUrl = () => {
    if (typeof window !== "undefined" && window.location) {
      let url = window.location.pathname
      if (url !== "/") {
        let paths = url.split("/")
        if (paths[1] === "blog") {
          return "/"
        } else {
          return `/${paths[1]}`
        }
      }
      return url
    }
  }

  isSelected = (url) => {
    return this.getRootUrl() === url ? "selected" : ""
  }

  render() {
    return (
      <nav
        className={`menubar ${this.props.active ? "active" : ""}`}
        onClick={this.props.toggle.bind(this)}
      >
        <ul>
          {navigation.map((item) => {
            return (
              <li
                key={`menubar-${item.url}`}
                className={`menu-item ${this.isSelected(item.url)}`}
              >
                <Link to={item.url}>{item.name}</Link>
              </li>
            )
          })}
        </ul>
      </nav>
    )
  }
}

export default Menubar