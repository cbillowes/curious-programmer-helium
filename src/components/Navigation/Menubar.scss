@import "../../layout/variables.scss";

@mixin flyout {
    opacity: 0;
    -webkit-transform: translate3d(300px, 0, 0);
    -ms-transform: translate3d(300px, 0, 0);
    transform: translate3d(300px, 0, 0);
    -webkit-transition: -webkit-transform 0.5s ease, opacity 0.3s ease 0.7s;
    transition: transform 0.5s ease, opacity 0.3s ease 0.7s;
}

@mixin flyin {
    opacity: 1;
    -webkit-transition: -webkit-transform 0.3s ease, opacity 0s ease 0s;
    transition: transform 0.3s ease, opacity 0s ease 0s;
    -webkit-transform: translate3d(0, 0, 0);
    -ms-transform: translate3d(0, 0, 0);
    transform: translate3d(0, 0, 0);
}

.navigation {
    .menubar {
        background-color: $background-darker;
        position: fixed;
        width: $menubar-width;
        top: 4em;
        right: 0;
        height: 100vh;
        z-index: 999;
        text-shadow: none;

        @include flyout;

        &.active {
            @include flyin;
        }

        ul {
            padding-left: 0;
            margin: 0;

            li.menu-item {
                list-style: none;

                &.selected {
                    background-color: $blue;

                    a {
                        color: $white;
                    }
                }

                a {
                    display: block;
                    padding: 0.25em 2em;
                    font-family: $font-family-sans-serif;
                    color: $foreground;

                    .github {
                        margin-top: 0.5em;
                        margin-left: 0.5em;

                        path {
                            fill: $foreground;
                        }
                    }

                    &:hover {
                        background-color: $primary;
                        color: $background-lighter;

                        .github path {
                            fill: $background;
                        }
                    }
                }
            }
        }
    }
}
