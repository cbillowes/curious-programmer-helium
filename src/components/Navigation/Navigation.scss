@import "../../layout/layout.scss";

.navigation {
  height: 4em;
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  background-color: $navigation-background;
  border-bottom: solid 10px $primary;
  z-index: 999;

  .normal-weight {
    font-weight: normal;
  }

  + div {
    margin-top: 6em;
  }
}