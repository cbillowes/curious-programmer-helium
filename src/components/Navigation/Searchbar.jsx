import React, { Component } from "react"
import "./Searchbar.scss"

class SearchBar extends Component {
  constructor(props) {
    super(props)
    this.state = {
      query: "",
      loaded: false,
      results: []
    }
    this.searchQuery = React.createRef()
  }

  generateResults = () => {
    if (this.state.error) {
      return (
        <li key="searchbar-empty" className="empty">
          Sorry, but something strange is going on. Search does not want to
          behave appropriately. Perhaps try again a little later. Drop me a line
          if it persists.
        </li>
      )
    }

    if (this.state.loaded && this.state.query !== "") {
      let numberOfResults = this.state.results.length
      if (numberOfResults === 0) {
        return (
          <li key="searchbar-empty" className="empty">
            Nothing came up. Go on, give it another shot.
          </li>
        )
      }

      let results = []
      for (let i = 0; i < numberOfResults; i++) {
        let result = this.state.results[i]
        let title = result.title
          .toString()
          .replace(/\| Curious Programmer/gi, "")
        let url = result.url
          .toString()
          .replace(/https:\/\/curiousprogrammer.io/gi, "")
        results.push(
          <li key={`searchbar-${i}`} onClick={this.props.toggle.bind(this)}>
            <a href={url}>{title}</a>
          </li>
        )
      }
      return results
    } else {
      return (
        <li key="searchbar-empty" className="empty">
          What do you feel like reading today? Search as you type, also, enter
          works too.
        </li>
      )
    }
  }

  query = (e) => {
    let query = this.searchQuery.current.value
    if (query.length > 4 || e.keyCode === 13) {
      if (query !== this.state.query) {
        this.search(query)
      }
    }

    if (query.length === 0) {
      this.setState({ query: "" })
    }
  }

  search = (query) => {
    fetch(
      `https://curiousprogrammer.tk/solr/oxygen/select?q=text:${encodeURIComponent(
        query
      )}&fl=url,title&rows=20&wt=json`
    )
      .then((res) => res.json())
      .then(
        (result) => {
          let results = []
          if (result.response) {
            results = result.response.docs
          }
          this.setState({
            query: query,
            loaded: true,
            results: results
          })
        },
        (error) => {
          this.setState({
            loaded: true,
            error: error
          })
        }
      )
  }

  render() {
    return (
      <div className={`searchbar ${this.props.active ? "active" : ""}`}>
        <input
          ref={this.searchQuery}
          className="search-query"
          type="text"
          onKeyUp={this.query.bind(this)}
        />
        <div className="results">
          <ul>{this.generateResults()}</ul>
        </div>
      </div>
    )
  }
}

export default SearchBar
