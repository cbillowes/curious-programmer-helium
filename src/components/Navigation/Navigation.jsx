import React, { Component } from "react"
import Logo from "./Logo/Logo"
import Icons from "./Icons/Icons"
import Menubar from "../Menubar/Menubar"
import Searchbar from "../Search/Searchbar"
import "./Navigation.scss"

class Navigation extends Component {
  constructor(props) {
    super(props)
    this.state = {
      menu: false,
      search: false
    }
  }

  toggleSearch = () => {
    this.setState({
      search: !this.state.search,
      menu: false
    })
  }

  toggleMenu = () => {
    this.setState({
      menu: !this.state.menu,
      search: false
    })
  }

  collapseAll = () => {
    this.setState({
      menu: false,
      search: false
    })
  }

  render() {
    return (
      <div className="navigation">
        <Logo onLogoClick={this.collapseAll} />
        <Icons
          onMenuClick={{ toggle: this.toggleMenu, active: this.state.menu }}
          onSearchClick={{ toggle: this.toggleSearch, active: this.state.search }}
        />
        <Menubar
          toggle={this.toggleMenu}
          active={this.state.menu} />
        {/* <Searchbar
          toggle={this.toggleSearch}
          active={this.state.search} /> */}
      </div>
    )
  }
}

export default Navigation
