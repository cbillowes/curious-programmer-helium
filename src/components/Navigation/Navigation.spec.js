import React from "react"
import renderer from "react-test-renderer"
import { mount } from "enzyme"
import Navigation from "./Navigation"

describe("a navigation component", () => {
  let element = <Navigation />
  let component
  
  beforeAll(() => {
    component = mount(element)    
  })

  it("should render", () => {
    const tree = renderer.create(element).toJSON()
    expect(tree).toMatchSnapshot()
  })
})