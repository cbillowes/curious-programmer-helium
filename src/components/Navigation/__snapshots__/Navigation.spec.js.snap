// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`a navigation component should render 1`] = `
<div
  className="navigation"
>
  <div
    className="logo"
  >
    <div
      className="default-logo"
    >
      <a
        aria-current="page"
        className=""
        href="/"
        onClick={[Function]}
        onMouseEnter={[Function]}
        style={Object {}}
      >
        { 
        <span
          className="normal-weight"
        >
          curious
        </span>
         
        programmer
         }
      </a>
    </div>
    <div
      className="mobile-logo"
    >
      <a
        aria-current="page"
        className=""
        href="/"
        onClick={[Function]}
        onMouseEnter={[Function]}
        style={Object {}}
      >
        { }
      </a>
    </div>
  </div>
  <div
    className="icons"
  >
    <div
      className="icon search "
      onClick={[Function]}
    >
      <i />
    </div>
    <div
      className="icon menu "
      onClick={[Function]}
    >
      <i />
    </div>
  </div>
  <nav
    className="menubar "
    onClick={[Function]}
  >
    <ul>
      <li
        className="menu-item selected"
      >
        <a
          aria-current="page"
          className=""
          href="/"
          onClick={[Function]}
          onMouseEnter={[Function]}
          style={Object {}}
        >
          Blog
        </a>
      </li>
      <li
        className="menu-item "
      >
        <a
          href="/about"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          About
        </a>
      </li>
      <li
        className="menu-item "
      >
        <a
          href="/tags"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          Tags
        </a>
      </li>
      <li
        className="menu-item "
      >
        <a
          href="/credits"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          Credits
        </a>
      </li>
      <li
        className="menu-item "
      >
        <a
          href="/privacy-policy"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          Privacy Policy
        </a>
      </li>
      <li
        className="menu-item "
      >
        <a
          href="/license"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          License
        </a>
      </li>
    </ul>
  </nav>
  <div
    className="searchbar "
  >
    <input
      className="search-query"
      onKeyUp={[Function]}
      type="text"
    />
    <div
      className="results"
    >
      <ul>
        <li
          className="empty"
        >
          What do you feel like reading today? Search as you type, also, enter works too.
        </li>
      </ul>
    </div>
  </div>
</div>
`;
